compute/cker/include/cker/operation/Reduce.h

```diff
@@ -39,19 +39,47 @@ inline bool ReduceImpl(const In *input_data, const Shape &input_shape, const Sha
   const auto input_num_dims = input_shape.DimensionsCount();
 
   // Reset input iterator.
-  for (int idx = 0; idx < input_num_dims; ++idx)
+  if (num_axis == 1 && (axis[0] == -1 || axis[0] == input_num_dims - 1))
   {
-    input_iter[idx] = 0;
+    int input_size = 1;
+    int reduce_size = 0;
+    for (int idx = 0; idx < input_num_dims - 1; idx++)
+    {
+      input_size *= input_dims[idx];
+    }
+    reduce_size = input_dims[input_num_dims - 1];
+    for (int idx = 0; idx < input_size; idx++)
+    {
+      for (int r_idx = 0; r_idx < reduce_size; r_idx++)
+      {
+        if (r_idx == 0)
+        {
+          output_data[idx] = input_data[idx * reduce_size];
+        }
+        else
+        {
+          output_data[idx] = reducer(output_data[idx], input_data[idx * reduce_size + r_idx]);
+        }
+      }
+    }
+    return true;
   }
-  // Iterate through input_data.
-  do
+  else
```

Comment:
@YiHyunJin I think we don't need `else` because the special case of `axis = -1` is already handled and early exited.